jerry-core/parser/js/js-parser-expr.c

```diff
@@ -222,7 +225,18 @@ parser_parse_array_literal (parser_context_t *context_p) /**< context */
     {
       if (pushed_items > 0)
       {
-        parser_emit_cbc_call (context_p, CBC_ARRAY_APPEND, pushed_items);
+#if ENABLED (JERRY_ES2015)
+        if (pushed_spread_element)
+        {
+          parser_emit_cbc_call (context_p, PARSER_TO_EXT_OPCODE (CBC_EXT_SPREAD_ARRAY_APPEND), pushed_items);
+        }
+        else
+        {
+#endif /* ENABLED (JERRY_ES2015) */
+          parser_emit_cbc_call (context_p, CBC_ARRAY_APPEND, pushed_items);
+#if ENABLED (JERRY_ES2015)
+        }
+#endif /* ENABLED (JERRY_ES2015) */
```

Comment:
These #if guards are a bit awkward.  What if we do this?:
```c
#if ENABLED(..)
if (pushed_spread_element)
{
   ... 
   return;
}
#endif

parser_emit_cbc_call (....)
```

Reply:
Or maybe we could guard only the second argument?
```c
cbc_opcode_t cbc_opcode = CBC_ARRAY_APPEND;

#if ENABLED(..)
if (pushed_spread_element)
{
   cbc_opcode = PARSER_TO_EXT_OPCODE (CBC_EXT_SPREAD_ARRAY_APPEND);
}
#endif

parser_emit_cbc_call (context_p, cbc_opcode, pushed_items);
```

This solution could also be used below when the same append is invoked. Maybe this could even be extracted into an inline method?